src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +86,136 @@ void VulnerabilityScannerFacade::start(
         auto scanOrchestrator = std::make_shared<ScanOrchestrator>(
             m_indexerConnector, m_databaseFeedManager, m_reportSocketClient, m_internalMutex);
 
+        // Queues to process skipped events
+        m_deltaQueue = std::make_shared<Utils::SafeQueue<std::string, RocksDBQueue<std::string>>>(
+            RocksDBQueue<std::string>(DELTA_EVENTS_QUEUE_PATH));
+        m_syncQueue = std::make_shared<Utils::SafeQueue<std::string, RocksDBQueue<std::string>>>(
+            RocksDBQueue<std::string>(SYNCHRONIZATION_EVENTS_QUEUE_PATH));
+
+        // Threads to retry skipped events
+        m_deltaThread = std::thread(
+            [scanOrchestrator, this]()
+            {
+                while (!m_stopRetryThread.load())
+                {
+                    for (int i = 0; i < SLEEP_RETRY_THREADS_SEC; ++i)
+                    {
+                        std::this_thread::sleep_for(std::chrono::seconds(1));
+                        if (m_stopRetryThread.load())
+                        {
+                            break;
+                        }
+                    }
+                    std::unique_lock<std::mutex> lock(m_retryDeltaMutex);
+                    m_retryDeltaCondition.wait(lock,
+                                               [this] { return !m_deltaQueue->empty() || m_stopRetryThread.load(); });
+                    if (!m_deltaQueue->empty())
+                    {
+                        std::shared_ptr<std::string> returnStr = m_deltaQueue->pop();
+                        std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*>
+                            data = SyscollectorDeltas::GetDelta(reinterpret_cast<const char*>(returnStr->c_str()));
+
+                        try
+                        {
+                            logDebug1(WM_VULNSCAN_LOGTAG, "Retry dbsync_packages event");
+                            scanOrchestrator->run(data);
+                        }
+                        catch (...)
+                        {
+                            logWarn(WM_VULNSCAN_LOGTAG, "Discarded dbsync_packages event");
```

Comment:
This was already discussed with the team and we did this to prevent the queue from having elements forever. Please @sebasfalcone validate this.

Reply:
Can we discern why the push failed? (I think is related to [this](https://github.com/wazuh/wazuh/pull/21456/files#r1464179107))

Reply:
Then this is a design decision.
Thank you.

Reply:
I added information about the event that failed and the cause of it. I had to create a specific flatbuffer and parse that information because it is not available there, only the variant is present at that moment